### PR TITLE
[WIP] Do not merge-- please review

### PR DESCRIPTION
**Pull request by cjschroder**
Please review for markup, spelling, tyops. I will add the remaining section after review.

### CODE REVIEW

**Comment by cjschroder** [comment]
This needs a reference that the first section is for the ML2 drivers.

**Comment by cjschroder** [comment]
Now here is the complete document.

File: xml/depl_cisco_opflex.xml [hunk 1]
```diff
@@ -0,0 +1,686 @@
...
+    <sect1 xml:id="sec.networking">
+  <title>Configure Networking on Compute Nodes</title>     
+    <para>
+In the example network we have two compute nodes that are connected to our Cisco ACI fabric. Network configuration will be automated in future releases, but for now you must follow these steps to configure networking on all of your compute nodes.
+    </para>
+    
+<procedure>
+    <step>
+        <para>
+            Enter <literal>YAST | System | Network Settings</literal> (Click Continue when you see the Chef client warning).
```
Comment (tbazant) [inline xml/depl_cisco_opflex.xml]: For menu items, we use <menuchoice><guimenu>&yast;</guimenu><guimenu>System</guimenu><guimenu>...</guimenu></menuchoice>

File: xml/depl_cisco_opflex.xml [hunk 2]
```diff
@@ -0,0 +1,686 @@
...
+   
+   <step>
+    <para>
+       Go to the General tab and set the MTU to 1600.
+    </para>
+   </step> 
+   
+    <step>
+    <para>
+       Press F10 to continue.
```
Comment (tbazant) [inline xml/depl_cisco_opflex.xml]: Use <keycap>F10</keycap> instead

File: xml/depl_cisco_opflex.xml [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+
+
+<para>
+    You should be back at the first screen at <literal>YAST | System | Network Settings</literal>. Press F3 (Add) to open the Hardware Dialog. 
+</para>
+
+  <procedure>
+   <step>
+    <para>
+      Under <literal>Device Type</literal> select VLAN.
```
Comment (tbazant) [inline xml/depl_cisco_opflex.xml]: For GUI elemets, use <guimenu>Device Types</guimenu>

File: xml/depl_cisco_opflex.xml [hunk 4]
```diff
@@ -0,0 +1,686 @@
...
+    <para>
+     Go back into Network Settings (Click Continue when you see the Chef client warning).
+    </para>
+   </step> 
+   
+   <step>
+    <para>
+        Change to the Routing tab and add these entries to the Routing Table:
+        
+        <command>Destination     224.0.0.0
```
Comment (tbazant) [inline xml/depl_cisco_opflex.xml]: For multi-line output, use <screen> instead of <command>

File: xml/depl_cisco_opflex.xml [hunk 5]
```diff
@@ -0,0 +1,686 @@
...
+
+<sect1 xml:id="sec.connectivity">
+  <title>Confirm Correct Connectivity</title>
+
+  <para>
+      Edit the <filename>/etc/dhclient.conf</filename> file on all compute nodes to verify that <literal>send dhcp-client-identifier</literal> has the MAC address of the NIC on the ACI fabric. (Run <command>ip addr show</command> to find the MAC address.) It should look like this, uncommented, with <literal>1:</literal> prepended to your MAC address, followed by a semi-colon, like this example: 
+   </para>
+   
+    <para> 
+        <literal>
```
Comment (tbazant) [inline xml/depl_cisco_opflex.xml]: Use <command> instead of literal here.

File: xml/depl_cisco_opflex.xml [hunk 6]
```diff
@@ -0,0 +1,686 @@
...
+        <literal>  
+     send dhcp-client-identifier 1:0:a0:24:ab:fb:9c;
+     </literal>
+  </para>
+  
+  <para>
+     Now install and run <command>lldpd</command>, the Link Layer Discovery Protocol daemon. You will use this to verify correct connectivity between your compute nodes.
+ </para>
+ 
+ <screen><prompt>root #  </prompt><command>zypper in lldpd</command>
```
Comment (tbazant) [inline xml/depl_cisco_opflex.xml]: For prompts, we use entities, such as &prompt.root;

File: xml/depl_cisco_opflex.xml [hunk 7]
```diff
@@ -0,0 +1,686 @@
...
+</para>   
+  
+<para>
+    <literal>
+    PortDescr: topology/pod-1/paths-102/pathep-[eth1/34]
+    </literal>
+</para> 
+
+<warning>
+    <title>Do not proceed if you do not see proper output from lldpctl</title>
```
Comment (tbazant) [inline xml/depl_cisco_opflex.xml]: we use capitalized <titles> -> <title>Do Not Proceed If You Do Not See Proper Output from <command>lldpctl</command></title>

File: xml/depl_cisco_opflex.xml (hunk 1 above)
Comment (sknorr) [inline xml/depl_cisco_opflex.xml]: Replace: `<literal>YAST | System | Network Settings</literal>` -> `<menuchoice><guimenu>YaST</guimenu><guimenu>System</guimenu><guimenu>Network Settings</guimenu></menuchoice>` (Yes, it is more markup.) Also, Continue should be in a `<guimenu>` as well.

File: xml/depl_cisco_opflex.xml [hunk 8]
```diff
@@ -0,0 +1,475 @@
...
+<dm:docmanager xmlns:dm="urn:x-suse:ns:docmanager">
+    <dm:maintainer>fs</dm:maintainer>
+    <dm:status>editing</dm:status>
+    <dm:deadline/>
+    <dm:priority/>
+    <dm:translation>no</dm:translation>
+    <dm:languages/>
+</dm:docmanager>
+</info>
+<para/>
```
Comment (sknorr) [inline xml/depl_cisco_opflex.xml]: Generally, add a real intro section here...

File: xml/depl_cisco_opflex.xml [hunk 9]
```diff
@@ -0,0 +1,475 @@
...
+    </para>
+
+   </listitem>
+  </itemizedlist>
+ </sect1>
+ 
+ <sect1 xml:id="sec.prereqisites">
+  <title>Prequisites</title>
+    <para>
+     As the Cisco ACI fabric is customized and highly-variable per customer's needs, you must have your ACI fabric already set up and working, and you need your own ACI expert administrator.
```
Comment (sknorr) [inline xml/depl_cisco_opflex.xml]: We usually use 79 character-lines.
You can try `daps-xmlformat -i SOURCEFILE.xml` to get our "official" style in an automated fashion.
(of course, be careful, if you apply this to established chapters, you may introduce far more changes than useful, but since this is new, it should be fine.)

File: xml/depl_cisco_opflex.xml [hunk 10]
```diff
@@ -0,0 +1,475 @@
...
+   
+   <step>
+    <para>
+        Highlight the Ethernet device connected to the ACI fabric and press F4 (Edit).
+    </para>
+   </step> 
+   
+    <step>
+    <para>
+        Enable <literal>No Link and IP Setup</literal>.
```
Comment (sknorr) [inline xml/depl_cisco_opflex.xml]: `literal` -> `guimenu`

File: xml/depl_cisco_opflex.xml (hunk 8 above)
Comment (fsundermeyer) [inline xml/depl_cisco_opflex.xml]: ...preferably as an abstract withing &lt;info>:

     <info>
      ...
      <abstract>
        <para>
          ABSTRACT
       </para>
     </abstract>
    </info>

File: xml/depl_cisco_opflex.xml (hunk 9 above)
Comment (fsundermeyer) [inline xml/depl_cisco_opflex.xml]: See http://vim.wikia.com/wiki/Automatic_formatting_of_paragraphs on how to configure vim to do this automatically

File: xml/depl_cisco_opflex.xml [hunk 11]
```diff
@@ -0,0 +1,686 @@
...
+    </para>
+   </step> 
+   
+   <step>
+    <para>
+       Go to the General tab and set the MTU to 1600.
```
Comment (fsundermeyer) [inline xml/depl_cisco_opflex.xml]: All UI elements need to be wrapped ion &lt;guimenu>, this also includes tab-titles. For values we use &lt;literal>:

    Go to the <guimenu>General</guimenu> tab and set the <guimenu>MTU</guimen> to <literal>1600</literal>.

File: xml/depl_cisco_opflex.xml (hunk 4 above)
Comment (fsundermeyer) [inline xml/depl_cisco_opflex.xml]: ...and make sure to never use &lt;screen> as an inline object__always use it as a block element.

    Wrong
    <para><screen>...</screen></para>

    Correct:
    <para>...</para>
    <screen>...</screen>

File: xml/depl_cisco_opflex.xml (hunk 5 above)
Comment (fsundermeyer) [inline xml/depl_cisco_opflex.xml]: I would rather use &lt;screen>, because this is a single command in a block element - execatly what we are using &lt;screen> for

File: xml/depl_cisco_opflex.xml (hunk 6 above)
Comment (fsundermeyer) [inline xml/depl_cisco_opflex.xml]: ... _not_ followed by  a space. Also there si no need to use command in screens:

    <screen>&prompt.root;zypper in lldpd</screen>

However, we are supposed to use sudo whenever possible (rather than requiring to log in as root:

       <screen>&prompt.user;sudo zypper in lldpd</screen>

_However_ using a prompt is usually only required if we want to display the command and its output, to make it possible to distinguis between the command and the output. To cut a long story short, the correct way to document these three commands is simply:

   <screen>sudo zypper in lldpd
    sudo systemctl start lldpd.service
    sudo lldpctl</screen>

File: xml/depl_cisco_opflex.xml [hunk 12]
```diff
@@ -0,0 +1,686 @@
...
+</para>
+
+<screen><prompt>root # </prompt><command>systemctl start openvswitch</command>
+<prompt>root # </prompt><command>systemctl enable openvswitch</command>
+<prompt>root # </prompt><command>ovs-vsctl add-br br-int</command>
+<prompt>root # </prompt><command>ovs-vsctl set bridge br-int protocols=[]</command>
+<prompt>root # </prompt><command>ovs-vsctl add-port br-int br-int_vxlan0 \
+    -- set Interface br-int_vxlan0 type=vxlan options:remote_ip=flow \
+    options:key=flow options:dst_port=8472</command>
+</screen>
```
Comment (fsundermeyer) [inline xml/depl_cisco_opflex.xml]: This is a lot of commands. Here you can use sudo -s to open a root shell that allows to enter the commands without having to type "sudo" at the beginning of each line:

    <screen>sudo -s
    systemctl start openvswitch
    ...
    ovs-vsctl add-port ...
    exit</screen>

File: xml/depl_cisco_opflex.xml [hunk 13]
```diff
@@ -0,0 +1,686 @@
...
+</para>
+</listitem>
+</itemizedlist>
+  
+  <para>
+  Now you will customize the Neutron barclamp for your installation. Note that the order of <literal>ml2_type_drivers</literal> is important, and <literal>opflex</literal> always comes first. The second <literal>ml2_type_drivers</literal> option defines the encapsulation used (<literal>vxlan</literal> or <literal>vlan</literal>).  Click on the Raw button to edit the attributes in raw mode and create a proposal. Set the following values and leave everything else set to the defaults:
+  </para>
+  
+  <screen>
+{
```
Comment (fsundermeyer) [inline xml/depl_cisco_opflex.xml]: To avoid unnecessary whitespace, always put the first line of the text directly behind the &lt;screen> tag and directly follow the last line with the closing tag:

    <screen>{
    ...
    }</screen>

File: xml/depl_cisco_opflex.xml [hunk 14]
```diff
@@ -0,0 +1,686 @@
...
+  </para>
+  
+  <para>
+      Now you can verify that the VLAN is correctly configured on the compute nodes, and the network interface has an IP address from the ACI fabric.
+  </para>
+  <screen><prompt>root # </prompt><command>ip a show dev vlan.4093</command>
+</screen> 
+<para>
+    If the interface does not get an IP address from the ACI fabric, see the troubleshooting section.
+</para>
```
Comment (fsundermeyer) [inline xml/depl_cisco_opflex.xml]: Is it possible to provide a link to the troubleshooting sections that deals with this problem?

File: xml/depl_cisco_opflex.xml [hunk 15]
```diff
@@ -0,0 +1,686 @@
...
+
+<tip><para>The vlan.4093 interface will not get an IP address from the fabric until Neutron is started on the control node (which was done in the previous step).  The driver looks at <literal>apic_provision_infra = True</literal> and <literal>apic_provision_hostlinks = True</literal> in <filename>ml2_conf_cisco_apic.ini</filename>to know that it must provision the ACI infrastructure with the compute host information.</para> </tip>
```
Comment (fsundermeyer) [inline xml/depl_cisco_opflex.xml]: Please make sure to always provide a title for an annotation.

File: xml/depl_cisco_opflex.xml (hunk 8 above)
Comment (cjschroder) [inline xml/depl_cisco_opflex.xml]: :) I used depl_cisco.xml as a template, so that is copied from it.

File: xml/depl_cisco_opflex.xml [hunk 16]
```diff
@@ -0,0 +1,686 @@
...
+    <para>
+       For further testing, remove the Dbaccess rule set from the Consumed Rule Set for the Wweb group to see the pings stop. Re-add the Dbaccess rule set to see the pings start again.
+    </para>
+   </step>
+   </procedure>
+ </sect1>
+ 
+ <sect1 xml:id="sec.troubleshooting">
+     <title>Troubleshooting</title>
+     <para>
```
Comment (fsundermeyer) [inline xml/depl_cisco_opflex.xml]: I suggest to turn this into a Q&A section. This allows to add more topics in the future and also makes it possible to link to each question. See xml/depl_troubleshooting.xml for a blueprint.

File: xml/depl_cisco_opflex.xml (hunk 13 above)
Comment (cjschroder) [inline xml/depl_cisco_opflex.xml]: Does whitespace matter to DAPS? It matters to me, as giant globs of unbroken text are hard to read.

Comment (sknorr) [inline xml/depl_cisco_opflex.xml]: White space matters in screens (and the other verbatim elements which we do not allow in GeekoDoc). These days, our stylesheets cut off initial and final whitespace.
However, it is best to not assume this sort of approach on the part of the stylesheets and instead make sure your screens do not contain extra whitespace.

File: xml/depl_cisco_opflex.xml (hunk 1 above)
Comment (cjschroder) [inline xml/depl_cisco_opflex.xml]: Holy cow, that is a lot of tags for a simple sentence.

Comment (cjschroder) [inline xml/depl_cisco_opflex.xml]: It look a lot nicer, but geez so many tags!

**Comment by cjschroder** [comment]
Hello it's me again :) Thank you very much for the detailed reviews. Please give this one more good review.

**Comment by cjschroder** [comment]
I mean "good" as in "thorough and nit-picky" :)

File: xml/depl_cisco_opflex.xml [hunk 17]
```diff
@@ -8,364 +8,411 @@
...
+ </info>
+ <sect1 xml:id="sec.deployment">
   <title>Deployment Into a Cisco ACI Fabric</title>
 
   <para>
- SUSE customers who wish to deploy &cloud; into a Cisco Application Centric Infrastructure (ACI) fabric environment can do this by using Cisco OpFlex ML2 or GBP drivers. Cisco ACI options are selected in the Neutron barclamp, which will make your &cloud; installation a member of your Cisco Application Policy Infrastructure Controller (APIC). The drivers are packaged as RPMs for easy installation.  <xref linkend="sec.barclamps"/> details how to use the ML2 drivers, and <xref linkend="sec.gbp"/> has the steps for using the GBP drivers.
- </para>
-   <para>
- This is a technology preview. We have successfully tested these steps on systems directly connected to a switch port on the ACI fabric. After you have set this up you should be able to:
+   SUSE customers who wish to deploy &cloud; into a Cisco Application Centric
```
Comment (sknorr) [inline xml/depl_cisco_opflex.xml]: We tend not to distance ourselves to much from our customers, hence: "SUSE customers who wish..." -> "If you wish..."

File: xml/depl_cisco_opflex.xml [hunk 18]
```diff
@@ -8,364 +8,411 @@
...
+   </listitem>
+  </itemizedlist>
+
   <para>
-  Now you will customize the Neutron barclamp for your installation. Note that the order of <literal>ml2_type_drivers</literal> is important, and <literal>opflex</literal> always comes first. The second <literal>ml2_type_drivers</literal> option defines the encapsulation used (<literal>vxlan</literal> or <literal>vlan</literal>).  Click on the Raw button to edit the attributes in raw mode and create a proposal. Set the following values and leave everything else set to the defaults:
+   Now you will customize the Neutron barclamp for your installation. Note that
+   the order of <literal>ml2_type_drivers</literal> is important, and
+   <literal>opflex</literal> always comes first. The second
+   <literal>ml2_type_drivers</literal> option defines the encapsulation used
+   (<literal>vxlan</literal> or <literal>vlan</literal>). Click on the
```
Comment (sknorr) [inline xml/depl_cisco_opflex.xml]: "Click on" -> "Click"

File: xml/depl_cisco_opflex.xml [hunk 19]
```diff
@@ -427,260 +474,337 @@ horizon
  }
 }
   </screen>
-  
+
   <para>
-      Save your changes and click Apply, and then apply the Neutron and Nova barclamps. If the Nova barclamp fails see <xref linkend="sec.troubleshooting"/>.
+   Save your changes and click Apply, and then apply the Neutron and Nova
```
Comment (sknorr) [inline xml/depl_cisco_opflex.xml]: guimenu for Apply?

File: xml/depl_cisco_opflex.xml [hunk 20]
```diff
@@ -427,260 +474,337 @@ horizon
...
-</screen> 
+  <tip>
+      <title>Getting an IP Address from the ACI Fabric</title>
+   <para>
+    The <replaceable>vlan.4093</replaceable> interface will not get an IP 
+address from the fabric until
+    Neutron is started on the control node (which was done in the previous
+    step). The driver looks at <literal>apic_provision_infra = True</literal>
+    and <literal>apic_provision_hostlinks = True</literal> in
+    <filename>ml2_conf_cisco_apic.ini</filename>to know that it must provision
```
Comment (sknorr) [inline xml/depl_cisco_opflex.xml]: Add a space between `</filename>` and `to`

File: xml/depl_cisco_opflex.xml [hunk 21]
```diff
@@ -427,260 +474,337 @@ horizon
...
+
+  <para>
+   Make the image public in the Web interface 
+(<menuchoice><guimenu>Admin</guimenu><guimenu>Images</guimenu><guimenu> Edit 
+Image</guimenu></menuchoice>, click the <guimenu>Edit Image</guimenu> button 
+and not the drop-down menu), and then create a new network 
+(<menuchoice><guimenu>Project</guimenu><guimenu>Network</guimenu><guimenu>
+Networks</guimenu></menuchoice>). Verify that the network has been created in 
+the APIC web interface under<guimenu>Tenants</guimenu>. Start the instances on 
+each compute node within the same project that are attached to the new network,
```
Comment (sknorr) [inline xml/depl_cisco_opflex.xml]: "within the same project that are attached..." -- Grammar seems off here.

File: xml/depl_cisco_opflex.xml [hunk 22]
```diff
@@ -427,260 +474,337 @@ horizon
...
-       </para>
-       
-       <screen>
+  </warning>
+
+  <para>
+   Start in the Crowbar Dashboard and edit the Neutron Barclamp. Change the
+   <literal>ml2</literal> driver to <literal>apic_gbp</literal> by highlighting
+   <literal>apic_gbp</literal>, then click <guimenu>Save</guimenu> and then 
+<guimenu>Apply.</guimenu>
```
Comment (sknorr) [inline xml/depl_cisco_opflex.xml]: Remove final punctuation from the guimenu.

File: xml/depl_cisco_opflex.xml [hunk 23]
```diff
@@ -427,260 +474,337 @@ horizon
...
+     and the DB instances.
     </para>
    </step>
-   
-  <step>
+   <step>
     <para>
-       For further testing, remove the Dbaccess rule set from the Consumed Rule Set for the Wweb group to see the pings stop. Re-add the Dbaccess rule set to see the pings start again.
+     For further testing, remove the Dbaccess rule set from the Consumed Rule
+     Set for the Web group to see the pings stop. Re-add the Dbaccess rule set
```
Comment (sknorr) [inline xml/depl_cisco_opflex.xml]: First sentence in this para is rather hard to read... what is the goal of our testing? why is "rule set" capitalized one time and in lowercase the other time? "to see the pings stop" sounds a bit awkward, especially when placed at the end (feels like an afterthought).

**Comment by sknorr** [comment]
I hope these are not too many comments... sorry. I also did not read too closely, so my apologies if I missed context.

**Comment by cjschroder** [comment]
Awesome sauce, thank you! There is no such thing as too many comments.

File: xml/depl_cisco_opflex.xml [hunk 24]
```diff
@@ -8,364 +8,411 @@
...
     </para>
-   </step> 
-   
-    <step>
+   </step>
+   <step>
     <para>
-     Press F10 to save and exit Network Settings, then press F9 to exit Yast.
+     Press <keycap>F10</keycap> to save and exit <guimenu>Network
+     Settings</guimenu>, then press <keycap>F9</keycap> to exit Yast.
```
Comment (cjschroder) [inline xml/depl_cisco_opflex.xml]: lol it's a nerd thing, weird spelling and capitalization.

**Comment by cjschroder** [comment]
Closing;  I will rebase and merge.
